in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,270 @@
+```
+
+The new `fee_asset_price_modifier_delta` is capped at (+/-) `MAXIMUM_FEE_ASSET_PRICE_PERCENT_CHANGE_PER_BLOCK` * `FEE_ASSET_PRICE_UPDATE_FRACTION` / 100.
+
+### `base_fee_wei_per_mana_congestion_multiplier`
+
+First we compute the excess mana in the current block by considering the parent mana spent and excess mana.
+
+```math
+\text{excess mana} = \begin{cases} 
+0 & \text{if } \text{parent.excess} + \text{parent.spent} < \text{TARGET MANA PER BLOCK} \\
+\text{parent.excess} + \text{parent.spent} - \text{TARGET MANA PER BLOCK} & \text{otherwise}
+\end{cases}
+```
+
+```math
+\begin{aligned}
+\text{base fee wei per mana congestion multiplier} &= \text{MINIMUM CONGESTION MULTIPLIER} * \text{exp}\left(\frac{\text{excess mana}}{\text{CONGESTION MULTIPLIER UPDATE FRACTION}}\right)
+\end{aligned}
+```
+
+## Deriving the base fee
+
+When a proposer is building a block, it calculates:
+```math
+\begin{aligned}
+\text{block cost in wei} &= \text{execution cost in wei} + \text{data cost in wei} \\
+\text{block cost in wei/mana} &= \frac{\text{block cost in wei}}{\text{TARGET MANA PER BLOCK}} + \text{proving cost wei per mana} \\
+\text{base fee in wei/mana} &= \text{block cost in wei/mana} * \text{base fee wei per mana congestion multiplier} \\
+\text{base fee in fee asset/mana} &= \text{base fee in wei/mana} * \text{fee asset per wei}
+\end{aligned}
+```
+
+This final value is the `base_fee_asset_per_mana` field in the L2 block header.
+
+## The cost of a transaction in the fee asset
+
+The amount of mana a transaction consumes is:
+
+```math
+\text{mana}_{tx} = \text{OVERHEAD MANA PER TX} + \sum_{op \in tx} \text{mana per operation}(op)
+```
+
+Therefore, the cost of a transaction's proposal, data publication, and verification at time $i$ is, denominated in the fee asset:
+
+```math
+\text{fee asset}_{tx} = \text{mana}_{tx} * \text{base fee asset per mana}
+```
+
+## Distribution of the fees
+
+When a transaction is included in a block in the pending chain, the L2 balance of the fee asset for the `fee_payer` of the transaction is reduced by the transactions's fee asset cost.
+
+When an epoch is proven, the fees for the epoch are paid out to each proposer for the block(s) they proposed.
+
+A percentage of each block's fee asset fees are paid to the prover of the epoch. The percentage is determined based on the quote submitted by the prover for the epoch.
+
+This quote is submitted by the prover in epoch `i+1` to prove epoch `i`, and is thus claimed/submitted to L1 by a proposer in epoch `i+1`.
+
+In order to prevent a prover bribing a proposer into accepting a quote with an extremely high fee, the maximum fee a prover can submit for an epoch proof quote is `MAXIMUM_EPOCH_PROOF_QUOTE_FEE`.
+
+## Clarifications and considerations
+
+### Why is `blobs_per_block` a constant?
+
+It makes computation easier. Otherwise block builder would need to know the number of blobs consumed by the transaction while it is building.
```

Comment:
Oh, this should've appeared earlier

Reply:
Wait, but blobs have a significant impact on the rollup's costs (or they will do once more people use blobs), so I'm not sure this makes sense to me.

Reply:
Please could you elaborate further on why it needs to be constant, to help me understand?
If I'm a block builder, I see a tx, I simulate it, I see how much data it wants to emit, I compute the L1 gas costs of that data (based on the proportion of blob(s) that this tx would use-up). I now know a rough estimate of how much ETH the data in this tx will cost the prover. Can we not then compute some approximation of this eth that the user can pay through L2 tokens?